recipes/openssl/3.x.x/conanfile.py

```diff
@@ -0,0 +1,617 @@
+        return "perl"
+
+    @property
+    def _nmake_makefile(self):
+        return r"ms\ntdll.mak" if self.options.shared else r"ms\nt.mak"
+
+    def _make(self):
+        with tools.chdir(self._source_subfolder):
+            # workaround for clang-cl not producing .pdb files
+            if self._is_clangcl:
+                tools.save("ossl_static.pdb", "")
+            args = " ".join(self._configure_args)
+            self.output.info(self._configure_args)
+
+            if self._use_nmake:
+                self._replace_runtime_in_file(os.path.join("Configurations", "10-main.conf"))
+
+            self.run('{perl} ./Configure {args}'.format(perl=self._perl, args=args), win_bash=self._win_bash)
+
+            self._patch_install_name()
+
+            self._run_make()
+
+    def _make_install(self):
+        with tools.chdir(self._source_subfolder):
+            self._run_make(targets=["install_sw"], parallel=False)
+
+    @property
+    def _cc(self):
+        if "CROSS_COMPILE" in os.environ:
+            return "gcc"
+        if "CC" in os.environ:
+            return os.environ["CC"]
+        if self.settings.compiler == "apple-clang":
+            return tools.XCRun(self.settings).find("clang")
+        elif self.settings.compiler == "clang":
+            return "clang"
+        elif self.settings.compiler == "gcc":
+            return "gcc"
+        return "cc"
+
+    def build(self):
+        with tools.vcvars(self.settings) if self._use_nmake else tools.no_op():
+            env_vars = {"PERL": self._perl}
+            if self.settings.compiler == "apple-clang":
+                xcrun = tools.XCRun(self.settings)
+                env_vars["CROSS_SDK"] = os.path.basename(xcrun.sdk_path)
+                env_vars["CROSS_TOP"] = os.path.dirname(os.path.dirname(xcrun.sdk_path))
+            with tools.environment_append(env_vars):
+                self._create_targets()
+                self._make()
+
+    @property
+    def _win_bash(self):
+        return tools.os_info.is_windows and \
+               not self._use_nmake and \
+            (self._is_mingw or tools.cross_building(self.settings, skip_x64_x86=True))
+
+    @property
+    def _make_program(self):
+        if self._use_nmake:
+            return "nmake"
+        make_program = tools.get_env("CONAN_MAKE_PROGRAM", tools.which("make") or tools.which("mingw32-make"))
+        make_program = tools.unix_path(make_program) if tools.os_info.is_windows else make_program
+        if not make_program:
+            raise Exception('could not find "make" executable. please set "CONAN_MAKE_PROGRAM" environment variable')
+        return make_program
+
+    def _patch_install_name(self):
+        if self.settings.os == "Macos" and self.options.shared:
+            old_str = '-install_name $(INSTALLTOP)/$(LIBDIR)/'
+            new_str = '-install_name '
+
+            tools.replace_in_file("Makefile", old_str, new_str, strict=self.in_local_cache)
```

Comment:
This failed in the build 12, seems like it was moved to a perl script. https://github.com/openssl/openssl/blame/7f81aed4bccbeb0ad491b68dd601d576c4956c78/Configurations/shared-info.pl#L46

Reply:
This is weird. The commit change this part is already part of 1.1.1

https://github.com/openssl/openssl/commit/793077d0beccfa20c9962546393128b92a7e68e4